### PR TITLE
Alchemy `predict --clean` improvements

### PR DESCRIPTION
**Pull request by JenkeScheen**
Adding:

- [x] extends `predict --clean` to also remove edges with `NaN` magnitudes
- [x] restrict `predict` to only do statistics with >3 experimental measurements; only 2 was giving stats regression errors
- [x] add multi-SDF writer by ways of `predict --wtop`: writes local multi-SDF of top `n` poses

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by JenkeScheen** [comment]
This is ready for a quick look over @hmacdope , I'm tackling some performance-related improvements in another PR: https://github.com/asapdiscovery/asapdiscovery/pull/1253

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py [hunk 1]
```diff
@@ -833,6 +833,14 @@ def stop(network_key: str):
     help="Make predictions using only the largest subnetwork present in the results. "
     "Useful in cases where the network is disconnected by e.g. simulation failures.",
 )
+@click.option(
+    "-wtop",
+    "--write-top-n-poses",
+    help="The number of top-scoring poses to write to a multi-SDF in the local directoy.",
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py:839]:
```suggestion
    help="The number of top-scoring poses to write to a multi-SDF in the local directory.",
```

Comment (hmacdope) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py:839]: Also why not just write them all? There isn't much of a difference space wise unless im missing something.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py [hunk 2]
```diff
@@ -936,6 +953,10 @@ def predict(
     )
     console.print(message)
 
+    # if requested, write an SDF of the top n compounds' docked poses
+    if write_top_n_poses > 0:
+        _ = get_top_n_poses(absolute_df, ligands, write_top_n_poses, console)
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/cli/alchemy.py:958]:
```suggestion
        _ = get_top_n_poses(absolute_df, ligands, write_top_n_poses, console, write_file=True)
```

Make the file writing side effect clearer.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py [hunk 3]
```diff
@@ -1046,21 +1048,64 @@ def clean_result_network(network, console=None):
             tf_res = results[0]
 
         deduped_results.append(tf_res)
-
     num_dupes_removed = len(cleaned_results) - len(deduped_results)
+
+    # remove predictions that have a NaN - this is extremely rare
+    denand_results = []
+    for result in deduped_results:
+        if not math.isnan(results[0].uncertainty.magnitude) and not math.isnan(
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py:1056]: why `results[0]` here?

**Review by hmacdope — CHANGES_REQUESTED** [review]
Some suggestions.

File: asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py (hunk 3 above)
Comment (JenkeScheen) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py:1056]: `results` is a tuple of results

File: asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py [hunk 4]
```diff
@@ -1046,21 +1048,64 @@ def clean_result_network(network, console=None):
             tf_res = results[0]
 
         deduped_results.append(tf_res)
-
     num_dupes_removed = len(cleaned_results) - len(deduped_results)
+
+    # remove predictions that have a NaN - this is extremely rare
+    denand_results = []
+    for result in deduped_results:
+        if not math.isnan(result.uncertainty.magnitude) and not math.isnan(
```
Comment (hmacdope) [inline asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py:1056]: This is now checking the same thing twice.

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py (hunk 4 above)
Comment (JenkeScheen) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py:1056]: lmao, ok, I'll dig back down just to check what I was initially querying here

Comment (JenkeScheen) [reply asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py]: ok fixed, the intention was to check for NaN in both the prediction and the uncertainty

**Review by hmacdope — APPROVED** [review]

**Pull request merged**